File: ios/Anemone SDK/model/config/SportWarning.swift
//
//  SportWarning.swift
//  Anemone SDK
//
//  Created by Angel Luis Garcia on 27/12/2017.
//  Copyright © 2017 Syltek Solutions S.L. All rights reserved.
//

import Foundation

public struct SportWarning {
    public let propertyId: PropertyId
    public let optionId: PropertyOptionId
    public let message: String
}

extension SportWarning: JSONMappable {
    public init(json: JSONObject) throws {
        self.propertyId = try PropertyId(json.getAny("property_id"))
        self.optionId = try PropertyOptionId(json.getAny("option_id"))
        self.message = try json.getString("message")
    }
}

File: android/app/src/main/java/com/playtomic/test/anemone_sdk/model/config/SportWarning.kt
package com.anemonesdk.model.config

import com.anemonesdk.general.json.JSONMappable
import com.anemonesdk.general.json.JSONObject
import org.json.JSONException

/**
 * Created by agarcia on 27/12/2017.
 */

class SportWarning : JSONMappable {
    val propertyId: PropertyId
    val optionId: PropertyOptionId
    val message: String

    constructor(propertyId: PropertyId, optionId: PropertyOptionId, message: String) {
        this.propertyId = propertyId
        this.optionId = optionId
        this.message = message
    }

    @Throws(JSONException::class)
    constructor(json: JSONObject) : super(json) {
        this.propertyId = PropertyId(json.getAny("property_id"))
        this.optionId = PropertyOptionId(json.getAny("option_id"))
        this.message = json.getString("message")
    }
}
